### ISSUE TITLE
Default config of no-unused-vars does not matches the docs

### ISSUE THREAD

**Issue by danielo515** [eslint/eslint#14016]
**Tell us about your environment**

Node version: v12.19.1
npm version: v6.14.8
Local ESLint version: v6.8.0 (Currently used)
Global ESLint version: Not found

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**
eslint-plugin-flowtype

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "extends": "react-app",
  "rules": {
    "no-console": "warn",
    "no-alert": "warn",
    "no-debugger": "warn",
    "react/jsx-curly-brace-presence": [2, "never"],
    "import/no-duplicates": ["error"],
    "import/no-namespace": ["error"],
    "import/newline-after-import": ["error"],
    "import/order": [
      "error",
      {
        "groups": ["builtin", "external", "parent", "sibling", "index"],
        "alphabetize": { "order": "asc" }
      }
    ]
  },
  "overrides": [
    {
      "files": ["src/**/*.spring.js"],
      "rules": {
        "react/jsx-key": "error",
        "flowtype/require-valid-file-annotation": ["error", "always"]
      }
    }
  ]
}
```

</details>

**What did you expect to happen?**
I was expecting to get warnings for unused vars on function calls, but eslint was not warning about it.
The default values I was expecting are documented here: https://github.com/eslint/eslint/blob/master/docs/rules/no-unused-vars.md#options

**What actually happened? Please include the actual, raw output from ESLint.**
The problem is that, instead of the documented default values, this is what I got out of eslint --print-config
```js
    "no-unused-vars": [
      "warn",
      {
        "args": "none",
        "ignoreRestSiblings": true
      }
    ]
```
I don't know what can be causing this default, but I didn't found any specific setup of that rule in any of my plugins.

**Comment by mdjermanovic** [comment]
Hi @danielo515, thanks for the issue!

You have `"extends": "react-app"` in the configuration.

[eslint-config-react-app](https://github.com/facebook/create-react-app/tree/master/packages/eslint-config-react-app) sets these options for the `no-unused-vars` rule [here](https://github.com/facebook/create-react-app/blob/0f6fc2bc71d78f0dcae67f3f08ce98a42fc0a57c/packages/eslint-config-react-app/index.js#L176).

**Comment by danielo515** [comment]
Dam it, I swear I tried any combination of ag, and grep that I can think of to find if anything on the node_modules was setting that rule, sorry for opening this issue then.
(by the way, will be nice if eslint provides a way to locate where certain rule is coming from)

Thanks for your quick answer.
